testcafe/tests/drive/folder_sharing_scenario.js

```diff
@@ -0,0 +1,134 @@
+import { Role } from 'testcafe'
+import { driveUser } from '../helpers/roles'
+import {
+  deleteLocalFile,
+  checkLocalFile,
+  setDownloadPath,
+  TESTCAFE_DRIVE_URL,
+  FOLDER_DATE_TIME
+} from '../helpers/utils'
+const homedir = require('os').homedir
+let data = require('../helpers/data')
+
+import DrivePage from '../pages/drive-model'
+const drivePage = new DrivePage()
+
+import PublicDrivePage from '../pages/drive-model-public'
+const publicDrivePage = new PublicDrivePage()
+
+//************************
+//Tests when authentified
+//************************
+fixture`Folder link Sharing Scenario`.page`${TESTCAFE_DRIVE_URL}/`.beforeEach(
+  async t => {
+    await t.useRole(driveUser)
+    await drivePage.waitForLoading()
+  }
+)
+
+test('Drive : Create a $test_date_time folder in Drive', async () => {
+  await drivePage.addNewFolder(FOLDER_DATE_TIME)
+  //We need to pass FOLDER_DATE_TIME through multiple fixture, so we cannot use ctx here.
+})
+
+test('Drive : from Drive, go in a folder, upload a file, and share the folder', async t => {
+  await drivePage.goToFolder(FOLDER_DATE_TIME)
+  await drivePage.openActionMenu()
+  await t.pressKey('esc') //close action Menu
+
+  await drivePage.uploadFiles([`../../data/${data.filePDF}`])
+  await drivePage.shareFolderPublicLink()
+
+  const link = await drivePage.copyBtnShareByLink.getAttribute('data-test-url')
+  if (link) {
+    console.log(`link : ` + link)
+    console.log('data.sharingLink -> ' + data.sharingLink)
+
+    data.sharingLink = link
+    console.log(`data.sharingLink : ` + data.sharingLink)
+  }
+})
+
+//************************
+// Public (no authentification)
+//************************
+fixture`Drive : Access a folder public link`.page`${TESTCAFE_DRIVE_URL}/`
+  .beforeEach(async t => {
+    await t.useRole(Role.anonymous())
+    await setDownloadPath(`${homedir()}/Downloads`)
+  })
+  .afterEach(async () => {
+    const filepath = `${homedir()}/Downloads/files.zip`
```

Comment:
we should have this `/Download/files.zip` in a constant somewhere with the PDF file etc.